app/src/main/java/de/westnordost/streetcomplete/settings/SettingsUtil.kt

```diff
@@ -0,0 +1,10 @@
+package de.westnordost.streetcomplete.settings
+
+import android.view.View
+import de.westnordost.streetcomplete.data.QuestType
+
+fun genericQuestTitle(resourceProvider: View, type: QuestType<*>): String {
+    val questTitleTemplate = resourceProvider.resources.getString(type.title)
+    val parameterCount = questTitleTemplate.split("%s").size - 1
+    return resourceProvider.resources.getString(type.title, *Array(parameterCount){"…"})
+}
```

Comment:
1. Better as first parameter `resources: Resources`
2. You can't assume that every placeholder is a `%s`. And I don't mean `%d` etc, but the `%4$2s` syntax which might rightfully occur sometimes. I looked at the [documentation for Formatter](https://developer.android.com/reference/java/util/Formatter.html) (which `Resources::getString()` uses internally) and sadly, it doesn't expose any functions with which one can find out how many placeholders are defined. Luckily, as far as I know, the Formatter is lenient in regards to if more args are supplied than there are placeholders.

So my suggestion is to just supply an array of 10 or so "…" and that's it.

The other possible solution would be to continue to allow optional strings for the arg array and let all the implementations of `getTitleArgs` behave like this...
```
    fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String?> {
        val name = tags["name"] ?: tags["brand"]
        return arrayOf(name) // <- name may be null
    }
```
... and use the `getTitleArgs` function for the display in the quest list as well, only replace every `null` with`...`. This may even be the cleanest solution.

Reply:
> So my suggestion is to just supply an array of 10 or so "…" and that's it.

That was my initial idea, but it seemed like an extremely ugly hack (though I would be deeply surprised to see it ever failing). But after thinking it again, assuming that all placeholders will be always `%s` is even uglier.

> ... and use the getTitleArgs function for the display in the quest list as well, only replace every null with"…". This may even be the cleanest solution.

OK, I will switch to this version.

Reply:
Looking at it now: only `OsmElementQuestType` has `getTitleArgs`, not `QuestType`, reuires passing `featureName: Lazy<String?>` and so on.

I think I will go with repeating dots 10 times.

Reply:
So, what is the problem? 99% of quests are OsmElementQuestTypes and featureName is just a Lazy<String?>, note the `?`, so it can also be a lazy that returns null